manager/manager.go

```diff
@@ -347,6 +359,35 @@ func (self *manager) Stop() error {
 	return nil
 }
 
+// initializeNVML initializes the NVML library and sets up the nvmlDevices map.
+func initializeNVML() {
+	if err := gonvml.Initialize(); err != nil {
+		glog.Warningf("GPU metrics would not be available. Could not initialize NVML: %v", err)
+		return
+	}
+	nvmlInitialized = true
+	numDevices, err := gonvml.DeviceCount()
+	if err != nil {
+		glog.Warningf("GPU metrics would not be available. Failed to get the number of nvidia devices: %v", err)
+		return
+	}
+	glog.Infof("NVML initialized. Number of nvidia devices: %v", numDevices)
+	nvidiaDevices = make(map[int]gonvml.Device, numDevices)
+	for i := 0; i < int(numDevices); i++ {
+		device, err := gonvml.DeviceHandleByIndex(uint(i))
+		if err != nil {
+			glog.Warningf("Failed to get nvidia device handle %d: %v", i, err)
+			continue
+		}
```

Comment:
This method is called only after NVML is initialized and we know the device count, I don't see how this can be a transient error after those two conditions are satisfied. The only failure scenario after that is if the user doesn't have permission to access the device ([documentation](http://docs.nvidia.com/deploy/nvml-api/group__nvmlDeviceQueries.html#group__nvmlDeviceQueries_1g52677ecb45f937c5005124608780a3f4)). In that case, there's no point in retrying.

Currently, the code requires the GPUs to be already attached and NVML installed before cadvisor is started. I am thinking of making it so that it rechecks for NVML/GPU presense every so often.